packages/api/src/api/contributions.ts

```diff
@@ -0,0 +1,233 @@
+import express, { Router } from 'express';
+import { env } from '../env';
+import { Contribution } from '../entities/Contribution';
+import { User } from '../entities/User';
+import logger from '../logger';
+import { Project } from '../entities/Project';
+
+export const contributions = Router();
+contributions.use(express.json());
+
+export interface PullRequestContribution {
+  id: string;
+  title: string;
+  permalink: string;
+  mergedAt: string;
+  author: {
+    login: string;
+  };
+}
+
+export interface Repository {
+  id: string;
+  nameWithOwner: string;
+}
+
+interface ContributionResponse {
+  data: {
+    search: {
+      pageInfo: {
+        hasNextPage: boolean;
+        endCursor: string | null;
+      };
+      nodes: PullRequestContribution[];
+    };
+  };
+}
+
+interface RepositoryResponse {
+  data: {
+    nodes: Repository[];
+  };
+}
+
+const buildProjectsQuery = async (projects: Project[]) => {
+  let idString = '';
+
+  projects.forEach((p) => {
+    idString += `"${p.nodeID}",`;
+  });
+
+  idString = idString.substring(0, idString.length - 1);
+
+  const repoQueryString = `
+    query { 
+      nodes(ids: [${idString}]){
+        ... on Repository {
+          id
+          nameWithOwner
+        }    
+      }
+    }
+  `;
+
+  const fetchRes = await fetch('https://api.github.com/graphql', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `bearer ${env.githubToken}`,
+    },
+    body: JSON.stringify({
+      query: repoQueryString,
+    }),
+  });
+
+  const { data: responseData }: RepositoryResponse = await fetchRes.json();
+
+  let queryString = '';
+
+  responseData.nodes.forEach((repo) => {
+    queryString += `repo:${repo.nameWithOwner} `;
+  });
+
+  return queryString;
+};
+
+const buildUsersQuery = (users: User[]) => {
+  let queryString = '';
+
+  users.forEach((u) => {
+    queryString += `author:${u.githubUsername} `;
+  });
+
+  return queryString;
+};
+
+const buildDateQuery = (startDate: Date, endDate: Date) => {
+  const locale = 'en-US';
+  const queryString = `${startDate.getFullYear()}-${(startDate.getMonth() + 1).toLocaleString(
+    locale,
+    { minimumIntegerDigits: 2 },
+  )}-${startDate
+    .getDate()
+    .toLocaleString(locale, { minimumIntegerDigits: 2 })}..${endDate.getFullYear()}-${(
+    endDate.getMonth() + 1
+  ).toLocaleString(locale, { minimumIntegerDigits: 2 })}-${endDate
+    .getDate()
+    .toLocaleString(locale, { minimumIntegerDigits: 2 })}`;
```

Comment:
Can we move these out to variables to make things a little more readable?

Reply:
done